homeassistant/components/atome/sensor.py

```diff
@@ -125,9 +124,22 @@ def update_live_usage(self):
                 self._is_connected,
                 self._subscribed_power,
             )
-
         except KeyError as error:
-            _LOGGER.error("Missing last value in values: %s: %s", values, error)
+            _LOGGER.error(
+                "Live Data : Missing last value in values: %s: %s", values, error
+            )
+            error_during_retrieve = True
+        return error_during_retrieve
```

Comment:
It is quite uncommon for a function to return `True` on failure and `False` on success. Flip this around?

I would also suggest just having a `return True` and `return False` and drop the unnecessary variable.

Reply:
Thanks for your feed back.
I will change the logic : return "success" instead of "error".
For the variable , I need to know whether the function has triggered an exception or not. Is there a classic way to do it (rather than to set a variable)

Reply:
I've change the return of functions
Simplify Day/Week/Month/Year function